plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -707,8 +707,8 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
       }
       if (vote_logger.is_enabled(fc::log_level::info) || _update_vote_block_metrics) {
          if (block->contains_extension(quorum_certificate_extension::extension_id())) {
-            if (const auto& active_finalizers = chain.head_active_finalizer_policy()) {
-               const auto& qc_ext = block->extract_extension<quorum_certificate_extension>();
+            const auto& qc_ext = block->extract_extension<quorum_certificate_extension>();
+            if (const auto& active_finalizers = chain.active_finalizer_policy(qc_ext.qc.block_num)) {
```

Comment:
It doesn't seem right to assume that `block` is on the head branch. We should probably use the branch that `block` is on, and fetch the `qc_ext.qc.block_num`  from that branch.

Reply:
`accepted_block` signal is on the head. We do have the id handy to find the correct branch, probably safer to do it that way.